### ISSUE TITLE
hyperopt does not follow sklearn base classes

### ISSUE THREAD

**Issue by TuKo** [IntelPNI/brainiak#95]
hyperopt implementation does not follow the implementation to be used as most algorithms in scikit-learn.

One should first define an object, e.g.:
optimizer = hpo(a=1, b=2, c=3) # initialization: setting algorithm parameters 
And then should call the fit() (or other) function:
optimizer.fit(data,labels,other_params)

Note: in the case of hyperopt the function name should be more a "minimize" than a "fit" and the object should receive the function to be minimized.

**Comment by mjanderson09** [comment]
Let's consider sklearn grid search as a template for how this interface should be implemented:

https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/grid_search.py

**Comment by TuKo** [comment]
I took a look at it, and I think it is really poor.
We should properly design base classes, taking the best from sklearn, but doing better in cases like this one.
